EVSL_1.0/SRC/dos_utils.c

```diff
@@ -293,16 +237,43 @@ int lsPol(const double *const intv, const int maxDeg, double (*ffun)(double),
       na = (ya[i] - yx[i]) / yx[i];
       nrm += fabs(na);
     }
-    if (nrm < tol) {
-      pol->deg = k + 1;
+    if (nrm < pol->tol) {
+      k++;
       break;
     }
```

Comment:
Sounds good to me. In the Matlab code the degree wasn't explicitly saved, so I did not think about deg = k - 1.

Should we realloc mu once we know the deg? I believe right now mu[deg:maxdeg] is unused, and I don't think we ever change mu later.